core/src/main/scala/org/apache/spark/sql/SnappyBaseParser.scala

```diff
@@ -396,8 +404,6 @@ object SnappyParserConsts {
   final val EXTENDED = nonReservedKeyword("extended")
   final val EXTERNAL = nonReservedKeyword("external")
   final val FULL = nonReservedKeyword("full")
-  final val FUNCTION = nonReservedKeyword("function")
-  final val FUNCTIONS = nonReservedKeyword("functions")
```

Comment:
This is a problem with identifier.
There is a syntax SHOW ALL FUNCTION where "ALL" is optional. So it can be SHOW FUNCTIONS <funcName> or SHOW ALL FUNCTIONS. In first case it fails and takes Key work FUNCTIONS as an identifier . Once made reserved keyword it works as identifiers ignores such literals.